### PR TITLE
Add tracing and spans

### LINKED ISSUE #18: Better Logging

### PR DESCRIPTION
**Pull request by NULLx76**
WIP: #18

### CODE REVIEW

**Comment by NULLx76** [comment]
I'm unsure if tracing fully understands the `#[hooks]` etc., so possibly the logging will improve when dropping the framework.

File: src/checks.rs [hunk 1]
```diff
@@ -17,45 +18,50 @@ pub async fn in_same_voice_check(
...
-            None => false,
-        })
-        .then(|| ())
-        .ok_or_else(|| {
-            SunnyError::user(
-                format!("I only take requests from users in {}", name.mention()).as_str(),
-            )
-        })?;
-    Ok(())
+    let span = span!(Level::INFO, "in_same_voice_check", ?msg);
```
Comment (Druue) [inline src/checks.rs:21]: Is there a reason why we're putting this here and wrapping it in an async move block instead of using the #[instrument]?

Comment (NULLx76) [reply src/checks.rs:21]: `#[instrument]` doesn't work with `#[check]` sadly, though if we upgrade Sunny to work without the whole framework this indeed wouldn't be necessary.

**Review by Druue — APPROVED** [review]

**Comment by Druue** [comment]
closes #18

**Pull request merged**